### PR TITLE
Adding controller layer

### PR DESCRIPTION
**Pull request by varunon9**
Issue https://github.com/gawdsnitkkr/sell-my-services-backend/issues/24 + added sellers and services APIs

### CODE REVIEW

File: controllers/search.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      /**
+       * A threshold = 0.3 gives a distance range ~ 47 KM
+       * since latitude and longitude can be negative so check 
+       * which one is greater after applying threshold
+       */
+      const latitudeMax = parseFloat(latitude) + config.latitudeThreshold;
+      const latitudeMin = parseFloat(latitude) - config.latitudeThreshold;
+      let latitudeRange = [latitudeMin, latitudeMax];
+      if (latitudeMin > latitudeMax) {
+        // swap values
```
Comment (AnshulMalik) [inline controllers/search.js]: We should move this latitude range calculation in a helper function, let's keep it for another PR.

File: controllers/search.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          return resolve([sellers, statusCode.OK]);
+        }).catch((err) => {
+          logger.error('controllers searchSellers', err);
+          return reject([
+            'Server side error', statusCode.INTERNAL_SERVER_ERROR
+          ]);
+        });
+    });
+  }
+};
```
Comment (AnshulMalik) [inline controllers/search.js]: missing new line, we should probably add this rule to eslint.

File: middlewares/index.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const jwt = require('jsonwebtoken'); // used to create, sign, and verify tokens
 const env = process.env.NODE_ENV || 'development';
 const config = require('../config/config.js')[env];
 
-const utilityService = require('../services/utilityService');
+const { saveRequestLog } = require('../services/utility');
```
Comment (AnshulMalik) [inline middlewares/index.js]: 👍

File: routes/auth/seller.js [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+    .then(([seller, responseCode]) => { 
+      res.status(responseCode)
+        .json({
+          success: true,
+          result: seller
+        });
+    }).catch(([err, responseCode]) => {
+      res.status(responseCode);
+      if (typeof(err) !== 'string') {
+        logger.error('routes /auth/sellers GET', err);
```
Comment (AnshulMalik) [inline routes/auth/seller.js]: I like this

File: routes/auth/seller.js [hunk 5]
```diff
@@ -27,18 +35,22 @@ router.get('/sellers', (req, res) => {
...
-      res.json({
-        success: true,
-        result: seller
-      });
-    }).catch(err => {
-      if (typeof(err) != 'string') {
-        console.error('Error /auth/seller/update/profile', err);
+  params.id = req.decoded.id;
+  params.email = req.decoded.email;
+  sellerController.updateSeller(params)
```
Comment (AnshulMalik) [inline routes/auth/seller.js]: Can we say `updateSeller(req.decoded)`?

File: routes/search.js [hunk 6]
```diff
@@ -3,11 +3,11 @@ const router = express.Router();
 
 const logger = require('../modules/logger');
 
-const { searchSellers } = require('../services/searchService');
+const searchController = require('../controllers/search');
```
Comment (AnshulMalik) [inline routes/search.js]: we can use destructuring here

File: services/README.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+A service assumes that all prior validations have been taken care of.
+It simply interacts with db to get data.
+
+We don't have `catch` block in services because it will be handled by corresponding controller and
+generating status code = 500 i.e. Internal Server Error.
```
Comment (AnshulMalik) [inline services/README.md]: 👍

File: services/validations/service.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+const models = require('../../models');
```
Comment (AnshulMalik) [inline services/validations/service.js]: we should name it `utils`?

Comment (AnshulMalik) [reply services/validations/service.js]: and probably keep both `doesSuchSellerExist ` and `doesSuchServiceExist ` in one file?

File: controllers/search.js (hunk 1 above)
Comment (varunon9) [reply controllers/search.js]: sure

File: controllers/search.js (hunk 2 above)
Comment (varunon9) [reply controllers/search.js]: sounds good

File: routes/auth/seller.js (hunk 5 above)
Comment (varunon9) [reply routes/auth/seller.js]: no params has all the fields e.g. new password. id and email cannot be updated so overriding here.

File: routes/search.js (hunk 6 above)
Comment (varunon9) [reply routes/search.js]: i thought since we'll be using all the controller functions of search, we better import whole module.

File: services/validations/service.js (hunk 8 above)
Comment (varunon9) [reply services/validations/service.js]: there might be more validations regarding seller and services in future so it kept it in separate files. Not sure if it's good idea. Your call.

File: routes/search.js (hunk 6 above)
Comment (AnshulMalik) [reply routes/search.js]: ok

File: services/validations/service.js (hunk 8 above)
Comment (AnshulMalik) [reply services/validations/service.js]: I think we should not complicate things for now. We can always separate them in future when there is enough complexity.

Comment (varunon9) [reply services/validations/service.js]: okay, fair enough.

**Comment by AnshulMalik** [comment]
We can do follow up changes in another PRs.
